mapcache.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2019 Ricardo Leite. All rights reserved.
+ * Licenced under the MIT licence. See COPYING file in the project root for
+ * details.
+ */
+
+#ifndef __MAPCACHE_H_
+#define __MAPCACHE_H_
+
+#include "log.h"
+#include "size_classes.h"
+#include <sys/mman.h>
+
+#define MAPCACHE_SIZE 64
+
+struct MapCacheBin {
+private:
+    char* _block = nullptr;
+    uint32_t _blockNum = 0;
+
+public:
+    char* Alloc();
```

Comment:
definitely need to write or mention somewhere that what comes through `Alloc` is then to be.. eventually given to `PageFree`

(or even better, add a `Dealloc()`/`Free()` that does exactly this?)

I mention this because I had the completely wrong idea of this PR, until I understood that this is really doing 1 `mmap` that then will eventually lead to N `munmap`.. and it is difficult to understand from the structure of the code

Reply:
Added Free to MapCache and added some comments to make it more explicit of what is happening.